tests/test_stl.cpp

```diff
@@ -237,6 +237,12 @@ TEST_SUBMODULE(stl, m) {
         .def("member_initialized", &opt_exp_holder::member_initialized);
 #endif
 
+#ifdef PYBIND11_HAS_FILESYSTEM
+    // test_fs_path
+    m.attr("has_filesystem") = true;
```

Comment:
nit (minor) Consider setting `m.attr("has_filesytem") = false;` in alternative branch.
Better to be explicit, then you simplify do `skipif(not m.has_filesystem)` and make the local contract clearer; otherwise, you could have `has_filesystem = False` and still admit the unittest.

Reply:
I was being consistent with all the other similar checks (has_optional, has_exp_optional, load_variant) which all use hasattr().  I agree just setting it to True/False looks cleaner, but that should rather be done globally in a separate cleanup PR?

Reply:
Great point! Will resolve this and make minor cleanup PR.